com.unity.render-pipelines.high-definition/Editor/Lighting/LightUnit/LightUnitSlider.cs

```diff
@@ -202,19 +240,141 @@ protected override float GetPositionOnSlider(float value)
             return ValueToSlider(value);
         }
 
-        protected override void DoSlider(Rect rect, SerializedProperty value, Vector2 range)
+        protected override void DoSlider(Rect rect, SerializedProperty value, Vector2 sliderRange)
         {
             value.floatValue = ExponentialSlider(rect, value.floatValue);
         }
 
         float ExponentialSlider(Rect rect, float value)
         {
-            var internalValue = GUI.HorizontalSlider(rect, ValueToSlider(value), 0f, 1f, GUI.skin.horizontalSlider, GUI.skin.horizontalSliderThumb);
-
+            var internalValue = GUI.HorizontalSlider(rect, ValueToSlider(value), 0f, 1f);
             return SliderToValue(internalValue);
         }
     }
 
+    /// <summary>
+    /// Formats the provided descriptor into a piece-wise linear slider with contextual slider markers, tooltips, and icons.
+    /// </summary>
+    class PiecewiseLightUnitSlider : LightUnitSlider
+    {
+        struct Piece
+        {
+            public Vector2 domain;
+            public Vector2 range;
+            public Func<float, float> transform;
+            public Func<float, float> inverseTransform;
+        }
+
+        // Piecewise function indexed by value ranges.
+        private readonly Dictionary<int, Piece> m_PiecewiseFunctionMap = new Dictionary<int, Piece>();
+
+        Func<float, float> GetTransformation(float x0, float x1, float y0, float y1)
```

Comment:
`GetTransformation` is not exactly what is does.

I would use instead `NewTransformation` or `CreateTransformation` because it instantiate a lambda function (thus creating GC allocation).